### ISSUE TITLE
Updating config with ArgumentParser removes args from the parser Namespace

### ISSUE THREAD

**Issue by radkoff** [wandb/client#1699]
**Describe the bug**
I'm using a yaml-based sweep to run a script in which I use ArgumentParser. I followed the config [instructions](https://docs.wandb.ai/ref/config) and called `update()`:

```
parser = ArgumentParser()
parser.add_argument('--foo', type=int)
args = parser.parse_args()
wandb.init(project="bar")
wandb.config.update(args)
print(args.foo)
```

In this example `foo` is controlled by the sweep, but I also have other arguments not controlled by the sweep (which take on default values), and I'd like for all arguments to be recorded under wandb's config. But in that last line, trying to access `args.foo` results in: `AttributeError: 'Namespace' object has no attribute 'foo'`. I get warnings about how the Config value `foo` was locked by 'sweep', which I understand and expect, but for some reason wandb removes the conflicting config values from the Namespace passed to `update()`, instead of just leaving them be. My workaround is to turn the args into a dict and copy them like so: `wandb.config.update(copy.copy(vars(args)))`

**Expected behavior**
Warnings are given that I'm updating a config controlled by a sweep, but those arg values remain in the `args` Namespace to be accessed by the training script like normal.

**Operating System**
 - OS: macOS `10.14.6`
 - Version: `0.10.13`

**Comment by ekorman** [comment]
I'm also having this problem on wandb version 0.10.13, but not on 0.10.12

**Comment by raubitsj** [comment]
Thank you for this report,  the behavior is a non intended side effect of some cleanups to help users avoid changing parameters that were already configured by the sweep.

We have reproduced the problem thanks to the code snippet and will make a development branch available within the next day and make a release with this fix in the next week (0.10.14)

For now, I recommend the workaround mentioned by @radkoff  (copying the args namespace) or using 0.10.12 until 0.10.14 is released.

**Comment by raubitsj** [comment]
Experimental fix for this issue:
```
pip install --upgrade git+git://github.com/wandb/client.git@fix/issue-1699
```

**Comment by raubitsj** [comment]
The fix was just released in 0.10.14:
```
pip install --upgrade wandb
```
